variables.tf

```diff
@@ -1,9 +1,444 @@
+  type        = number
+  default     = 365
+  validation {
+    condition     = var.retention_maximum > 0 && var.retention_maximum < 365243
+    error_message = "The specified duration for retention maximum period is not a valid selection!"
+  }
+}
+
+variable "retention_minimum" {
+  description = "Specifies minimum duration of time an object must be kept unmodified for COS bucket. Only used if 'create_cos_bucket' is true."
+  type        = number
+  default     = 1
+  validation {
+    condition     = var.retention_minimum > 0 && var.retention_minimum < 365243
+    error_message = "The specified duration for retention minimum period is not a valid selection!"
+  }
+}
+
+variable "retention_permanent" {
+  description = "Specifies a permanent retention status either enable or disable for COS bucket. Only used if 'create_cos_bucket' is true."
+  type        = bool
+  default     = false
+}
+
+variable "object_versioning_enabled" {
+  description = "Enable object versioning to keep multiple versions of an object in a bucket. Cannot be used with retention rule. Only used if 'create_cos_bucket' is true."
+  type        = bool
+  default     = false
+}
+
+variable "archive_days" {
+  description = "Specifies the number of days when the archive rule action takes effect. Only used if 'create_cos_bucket' is true. This must be set to null when when using var.cross_region_location as archive data is not supported with this feature."
+  type        = number
+  default     = null
+}
+
+variable "archive_type" {
+  description = "Specifies the storage class or archive type to which you want the object to transition. Only used if 'create_cos_bucket' is true."
+  type        = string
+  default     = "Glacier"
+  validation {
+    condition     = contains(["Glacier", "Accelerated"], var.archive_type)
+    error_message = "The specified archive_type is not a valid selection!"
+  }
+}
+
+variable "expire_days" {
+  description = "Specifies the number of days when the expire rule action takes effect. Only used if 'create_cos_bucket' is true."
+  type        = number
+  default     = null
+}
+
+variable "activity_tracker_crn" {
+  type        = string
+  description = "Activity tracker crn for COS bucket (Optional)"
+  default     = null
+  validation {
+    condition     = var.activity_tracker_crn != null ? can(regex("crn:v1:bluemix:public:logdnaat:(au-syd|eu-gb|eu-es|us-south|eu-de|us-east):a/[0-9a-f]{32}:[0-9a-fA-F]{8}-([0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}::", var.activity_tracker_crn)) : true
+    error_message = "Must be a valid activity tracker crn"
+  }
+}
+
+variable "sysdig_crn" {
+  type        = string
+  description = "Cloud Monitoring crn for COS bucket (Optional)"
+  default     = null
+}
+
+##############################################################################
+# COS bucket encryption variables
+##############################################################################
+
+variable "existing_kms_instance_guid" {
```

Comment:
I would ask for the CRN instead of the GUID and then parse the GUID from the CRN. It means you can then support taking in a KMS instance from a different account. 
To support that use case you need to use an alias provider for the KMS calls, and an optional KMS apikey. For reference, see https://github.com/terraform-ibm-modules/terraform-ibm-secrets-manager/pull/147